engine/classes/Elgg/Database/Plugins.php

```diff
@@ -24,28 +27,34 @@
 class Plugins {
 
 	/**
-	 * @var string[] Active plugin IDs with IDs as the array keys. Missing keys imply inactive plugins.
+	 * @var string[] Active plugins, with plugin ID => GUID. Missing keys imply inactive plugins.
 	 */
-	protected $active_ids = array();
+	protected $active_guids = array();
 
 	/**
-	 * @var bool Has $active_ids been populated?
+	 * @var bool Has $active_guids been populated?
 	 */
-	protected $active_ids_known = false;
+	protected $active_guids_known = false;
 
 	/**
-	 * @var \Elgg\Cache\MemoryPool
+	 * @var MemoryPool
 	 */
 	protected $plugins_by_id;
 
 	/**
+	 * @var PluginSettingsCache
+	 */
+	protected $settings_cache;
+
+	/**
 	 * Constructor
 	 *
-	 * @param \Elgg\EventsService    $events Events service
-	 * @param \Elgg\Cache\MemoryPool $pool   Cache for referencing plugins by ID
+	 * @param EventsService $events Events service
+	 * @param MemoryPool    $pool   Cache for referencing plugins by ID
```

Comment:
Needs phpdoc for PluginSettingsCache.

Also, bummer that we need to inject two caches? Can we pare that down to one?